fvm/transactionInvoker.go

```diff
@@ -88,7 +88,10 @@ func (i *TransactionInvoker) Process(
 
 		}
 
-		err := sth.Commit(nestedTxnId)
+		var err error
```

Comment:
This guard won't be needed after #3144, which handles the limit error more generally (but the test in this PR is still useful)